spec/custom_matchers.rb

```diff
@@ -0,0 +1,12 @@
+RSpec::Matchers.define :belong_to do |expected|
+  match { |actual| (actual.to_a.uniq - expected.to_a.uniq).empty? }
+end
+
+RSpec::Matchers.define :contain_from do |expected|
+  match do |actual|
+    actual = actual.to_a.map { |x| x.data.to_a }.flatten.uniq
+    expected.map!(&:values) unless expected.first.is_a? Array
+    expected = expected.flatten.uniq
+    (actual - expected).empty?
+  end
+end
```

Comment:
In line 8 : If the expected data is an array of hashes, it turns it into a 2D array. So, the DataFrame's 2D `data` is expected to be a subset of the expected 2D array. That way, columns are checked to be in order, but rows aren't (mapping isn't lost though).

For examples, when only some keys are matched for -

Ideal case 2D output, if keys were inserted in given order : `[[1,2,3],[4,5,6],[7,8,9]]`
Actual 2D output due to redis' insertion order of keys (say): `[[4,5,6],[7,8,9], [1,2,3]]`
Expected superset-2D output by RSpec : `[[1,2,3],[4,5,6],[7,8,9],[10,11,12],...]`

`(Actual-Expected)`
=> `[]`